### PR TITLE
Fixed parameter generation and added SQL query helper

### PR DESCRIPTION
**Pull request by spinx**
Not exactly sure why your parameter generation didn't work for me, but I've added a rather hackinsh solution for now. 

Also added showQuery method that you may want to use. 

Previously all parameters were named p1, because $parameters->containsKey($parameterName) didn't work as intended.

### CODE REVIEW

File: lib/Weavora/Doctrine/ORM/EntityQueryBuilder.php [hunk 1]
```diff
@@ -205,20 +205,58 @@ public function fetchScalar($parameters = array())
...
     }
+
+    /**
+     * Get parameters that are already used
+     *
+     * @return array
+     */
+    private function getUsedParameterNames(){
+        $parameters = $this->getParameters();
+        $names = [];
```
Comment (yurytolochko) [inline lib/Weavora/Doctrine/ORM/EntityQueryBuilder.php:231]: could you pls change this to $names = array() to support php 5.3

File: lib/Weavora/Doctrine/ORM/EntityQueryBuilder.php [hunk 2]
```diff
@@ -205,20 +205,58 @@ public function fetchScalar($parameters = array())
...
 
         return $parameterName;
     }
+
+    /**
+     * Get parameters that are already used
+     *
+     * @return array
+     */
+    private function getUsedParameterNames(){
```
Comment (yurytolochko) [inline lib/Weavora/Doctrine/ORM/EntityQueryBuilder.php:229]: could you use PSR-2 code convention:

``` php
private function getUsedParameterNames()
{
   ...
}
```

File: lib/Weavora/Doctrine/ORM/EntityQueryBuilder.php [hunk 3]
```diff
@@ -205,20 +205,58 @@ public function fetchScalar($parameters = array())
...
+        }
+        return $names;
+    }
+
+    /**
+     * Get SQL query with parameters
+     *
+     * @return string
+     */
+    function showQuery()
```
Comment (yurytolochko) [inline lib/Weavora/Doctrine/ORM/EntityQueryBuilder.php:258]: 1. missed public modifier: `public function showQuery`
2. it should be named like `getQueryPreview` cause it doesn't show anything, just tries to build query preview in very simplified mode without params quoting and etc.
3. it wouldn't work for named params like 'SELECT \* FROM table WHERE id = :id'

Guess we should add this method in separate pull request with fixes and **unit tests**

**Comment by yurytolochko** [comment]
@spinx thnx for contributing. Please checkout my notes above. Let me know if you have no possibility to apply fixes - I will do it myself then.

**Comment by spinx** [comment]
@yurytolochko As I said it's really hackish for now, but I do intend to build on this so that you can merge, I just opened a PR so that I'm now forced to follow through :) Will finish over the weekend.

**Pull request closed without merging**